pkg/glusterfs/controllerserver.go

```diff
@@ -56,16 +97,21 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
 
 	volSizeMB := int(utils.RoundUpSize(volSizeBytes, 1024*1024))
 
-	// Get Volume name : TODO use the values from request
-	volumeName := req.Name
-	glusterVol := req.GetParameters()["glustervol"]
-	glusterServer = req.GetParameters()["glusterserver"]
-	glusterURL := req.GetParameters()["glusterurl"]
-	glusterURLPort := req.GetParameters()["glusterurlport"]
-	glusterUser := req.GetParameters()["glusteruser"]
-	glusterUserSecret := req.GetParameters()["glusterusersecret"]
+	// parse the request.
+	parseResp, parseErr := cs.ParseRequest(req)
+	if parseErr != nil {
+		return nil, status.Error(codes.InvalidArgument, "failed to parse request")
+	}
+
+	if parseResp != nil && parseResp.gdVolReq != nil {
+		if len(parseResp.gdVolReq.Name) == 0 {
+			return nil, status.Error(codes.InvalidArgument, "volumename is nil")
+		}
+	} else {
+		return nil, status.Error(codes.InvalidArgument, "parse response in nil")
+	}
 
-	glog.V(3).Infof("Request fields:[ %v %v %v %v %v %v]", glusterVol, glusterServer, glusterURL, glusterURLPort, glusterUser, glusterUserSecret)
```

Comment:
Did this go away on purpose?

Reply:
Yes, I removed it deliberately. it was rather a debugging entry.